### PR TITLE
add TLS policy and timeouts, updated ext proc to read certs

### PR DESCRIPTION
**Pull request by kaushikmitr**
Added TLS policy as per: https://gateway.envoyproxy.io/latest/tasks/extensibility/ext-proc/ and updated extproc to read the certs. 
Added http, cluster, and ext proc timeouts in the yamls.

### CODE REVIEW

File: pkg/manifests/enable_patch_policy.yaml [hunk 1]
```diff
@@ -4,10 +4,6 @@ metadata:
   name: envoy-gateway-config
   namespace: envoy-gateway-system
 data:
-# This manifest's main purpose is to set `enabledEnvoyPatchPolicy` to `true`.
```
Comment (liu-cong) [inline pkg/manifests/enable_patch_policy.yaml:7]: let's keep the comments? they look useful

File: pkg/manifests/gateway.yaml [hunk 2]
```diff
@@ -31,17 +33,21 @@ spec:
...
+    - name: llm-inference-gateway
       sectionName: llm-gw
   rules:
   - backendRefs:
       - group: gateway.envoyproxy.io
         kind: Backend
-        name: backend-dummy
\ No newline at end of file
+        name: backend-dummy
+    timeouts:
```
Comment (liu-cong) [inline pkg/manifests/gateway.yaml:51]: nit: Add a comment why this is needed, and link the envoy doc

File: pkg/manifests/patch_policy.yaml [hunk 3]
```diff
@@ -26,13 +26,23 @@ spec:
           original_dst_lb_config:
             use_http_header: true
             http_header_name: "target-pod"
-          connect_timeout: 6s
+          connect_timeout: 3599s
           lb_policy: CLUSTER_PROVIDED
           dns_lookup_family: V4_ONLY
+          circuit_breakers:
```
Comment (liu-cong) [inline pkg/manifests/patch_policy.yaml:32]: Add a comment on what this does and why it's needed.

File: pkg/manifests/ext_proc.yaml [hunk 4]
```diff
@@ -1,69 +1,154 @@
...
+              - key: tls.crt
+                path: server.crt
+              - key: tls.key
+                path: server.key
 ---
 apiVersion: v1
 kind: Service
 metadata:
-  name: instance-gateway-ext-proc
+  name: grpc-ext-proc
```
Comment (liu-cong) [inline pkg/manifests/ext_proc.yaml:86]: I prefer the old name. grpc doesn't describe the purpose - all ext procs are grpc anyway.

File: pkg/manifests/ext_proc.yaml [hunk 5]
```diff
@@ -1,69 +1,154 @@
...
     metadata:
       labels:
-        app: instance-gateway-ext-proc
+        app: grpc-ext-proc
     spec:
       containers:
-      - name: instance-gateway-ext-proc
-        image: <BUILT-IMAGE>
+      - name: grpc-ext-proc
+        image: us-docker.pkg.dev/kaushikmitra-gke-dev/kaushikmitra-docker-repo/llm-ig-pp-ext-proc
```
Comment (liu-cong) [inline pkg/manifests/ext_proc.yaml:56]: let's only update this once we have a public image

File: pkg/ext-proc/main.go [hunk 6]
```diff
@@ -31,6 +34,7 @@ var (
 
 	refreshPodsInterval    = flag.Duration("refreshPodsInterval", 10*time.Second, "interval to refresh pods")
 	refreshMetricsInterval = flag.Duration("refreshMetricsInterval", 50*time.Millisecond, "interval to refresh metrics")
+	certPath               = ""
```
Comment (liu-cong) [inline pkg/ext-proc/main.go:37]: you can just declare it as a flag.String here

File: pkg/ext-proc/main.go [hunk 7]
```diff
@@ -82,16 +116,57 @@ func main() {
...
+
+	// Create a new credentials object
+	creds := credentials.NewTLS(&tls.Config{Certificates: []tls.Certificate{certificate}})
+
+	return creds, nil
+}
 
+func loadCA(caPath string) (*x509.CertPool, error) {
+	ca := x509.NewCertPool()
+	caCertPath := "server.crt"
```
Comment (liu-cong) [inline pkg/ext-proc/main.go:159]: why do we hard code it? Can we just make it a flag and default to "server.crt"?

File: pkg/ext-proc/main.go [hunk 8]
```diff
@@ -82,16 +116,57 @@ func main() {
...
-		os.Exit(0)
+		err = s.Serve(lis)
+		if err != nil {
+			log.Fatalf("failed to serve: %v", err)
+		}
 	}()
 
-	s.Serve(lis)
+	http.HandleFunc("/healthz", healthCheckHandler)
+	err = http.ListenAndServe(":8080", nil)
```
Comment (liu-cong) [inline pkg/ext-proc/main.go:127]: the handler is nil currently, fix it? Should we make the port configurable?

File: pkg/manifests/ext_proc.yaml [hunk 9]
```diff
@@ -1,69 +1,154 @@
...
...M1JjClk3empVQVVTN2JWV1ZGNEh1UlVSSEVmVlBpb2Nvcy9MUS81a3ArejFnbSt4ZjNsOTRiVlRxZDJOWEkyRHMxY0kKMHBEeDNsVkhaU0V5QjVvQlBFbVBlQVBpMXQyL2JmaG9DMkF0YXZDeDVOWkEzVHdHdlBhcUFMbmxPcUVLQ3F6dApqUU1qcEppQUQ1cXZhOWVFcXhCMEVmeEVTd0tCZ1FDOEZJT014V1E3WXJpQTZQUHloSDV1eUJic21ZZEhzWXlqCm9QTkVzSUduMW1KbWh6YUFpREM2MWFJZHEzU2ZYYUNvajhQUkRnb04za1g4Z0E0U2prdmRNY2VJMDdaQm45MEsKZFZEdldIV3ZEa2FDZjZINEt6d05BNytxc3FtY3hIVklPNFhOQnhLaEU2Z1FPWDJUZkhTVjM0cVVmdXpZQUdyWQpXU0FIVzR1NUp3S0JnQ2crUUo0SDFlMHZOc1RlVWNqTnFMb05pSWdiTnY5VTJTUmRjeHROS0MxME5Cd2EwTDlwCnNSNWlwa3R6cmR3S216VkU0VFVZd2JwTlpoSVlheW4vbCt1YTBpK0lkaWZ6Wi8rTG0wMkhJWTJDejdMekZuMW0Kc1JBRUk1NWlnMGtxQUlLUU1VajFEc2JvV1RPRVJLSWgyZUhzZEl3cXlhMWxKNU83M0xCeWg3RXhBb0dCQUlXdQp0d281eTFLQ2lzM2x4bThjU1ptSVV2TDg5VEM2UEZLWnRnK2V2YW1FTWhEYURwMHhNQ0c4Y3l4UGorY3ViMkVnClBuaTdWOTRmblBNcU9kWnFtZld3eWppdzhweVdlbEJTcnFKUTUrVHphcDZiQlk4VmUrdHNQTEdocC9rcmtva3kKOVptVGEyUmVTbGl0NUZvT1hmZWhHaWtPUmw0SmZlZ2xBU1Q1cHNpRkFvR0FOc1hSWGZBSGZkNGZlZWtPek9LNwppR3FmNHovTlFaZHlCWGNKZlRhdUpLak1XSGxvSUN2ZjdyblM0YXVEcUE1bmoxTy9Eam5tY01lT2QyNW4yakVlCk1ibU1DMnR1U2VRSW02NjNoaEFYaUk0L2w5SU5Nc3hRUDlzUEJQUjg4Z1RFK3lCTUsvTFY4Mm9vd1BvOG1XNnkKWDdESlJSK3dJY3RueXh6bVBQOExRaE09Ci0tLS0tRU5EIFBSSVZBVEUgS0VZLS0tLS0K
```
Comment (liu-cong) [inline pkg/manifests/ext_proc.yaml:9]: let's not put private key

File: pkg/manifests/ext_proc.yaml [hunk 10]
```diff
@@ -1,69 +1,154 @@
+---
+apiVersion: v1
```
Comment (liu-cong) [inline pkg/manifests/ext_proc.yaml:2]: Can you update README on how to generate the TLS certs and configure ext proc?

**Comment by ahg-g** [comment]
Do we still need this?

**Pull request closed without merging**